src/app/containers/ConsentBanner/index.canonical.test.jsx

```diff
@@ -128,7 +128,7 @@ describe('Canonical Consent Banner Container', () => {
     expect(Banner).toHaveBeenCalledWith(
       {
         onAccept: expect.any(Function),
-        onReject: cookieOnReject,
+        onReject: expect.any(Function),
```

Comment:
I think there was some test logic in here previously to test the focus behaviour, have you decided to cover that in the header tests: src/app/containers/Header/index.test.jsx ?

Just seems less than ideal that if someone modified logic in here it could break tests in the seemingly unrelated header component? The tests in the header are good but I wonder if there should be specific tests in this component for the focus logic, decoupled from the banner itself? I feel like there was in an earlier iteration of this PR?

Reply:
Yeah I decided to put them in the Header since the ConsentBanner is used purely within the Header, and since it also also creates the `brandRef` ref, the complexity of the tests is massively reduced. I could add a test here that would check if focus is pulled to the ConsentBanner header?

Reply:
You _could_ have tests a test that covers focusing on whatever is provided for `onDismissFocusRef`, this is distinct from the header tests as they are more 'integration tests' checking the whole approach links together imo. It would test that this functionality would work focusing on any ref you passed in, not just the header link.

Totally up to you, it may well be overkill as it is covered in the header compenent tests.